app/src/main/java/org/p2p/wallet/history/db/dao/TransactionDaoDelegate.kt

```diff
@@ -0,0 +1,57 @@
+package org.p2p.wallet.history.db.dao
+
+import org.p2p.wallet.history.db.entities.CloseAccountTransactionEntity
+import org.p2p.wallet.history.db.entities.CreateAccountTransactionEntity
+import org.p2p.wallet.history.db.entities.RenBtcBurnOrMintTransactionEntity
+import org.p2p.wallet.history.db.entities.SwapTransactionEntity
+import org.p2p.wallet.history.db.entities.TransactionEntity
+import org.p2p.wallet.history.db.entities.TransferTransactionEntity
+import org.p2p.wallet.history.db.entities.UnknownTransactionEntity
+import org.p2p.wallet.utils.findInstance
+import timber.log.Timber
+
+class TransactionDaoDelegate(
+    private val transactionDao: List<TransactionDao<*>>
```

Comment:
Btw, why we just don't call this one as repository?

Reply:
can't answer you that in two words.
let's do a call sometime and i'll explain

Reply:
I've explaned it, closing